### ISSUE TITLE
Travis CI make cppcheck has errors on macOS

### ISSUE THREAD

**Issue by gaul** [s3fs-fuse/s3fs-fuse#713]
Maybe a newer version there:

```
$ make cppcheck
cppcheck --quiet --error-exitcode=1 \
            --inline-suppr \
            --std=c++03 \
            -U CURLE_PEER_FAILED_VERIFICATION \
            -U P_tmpdir \
            -U ENOATTR \
            --enable=all \
            --suppress=missingIncludeSystem \
            --suppress=unusedFunction \
            --suppress=variableScope \
            src/ test/
[src/curl.cpp:4075]: (error) Memory leak: new_item
[src/gnutls_auth.cpp:154]: (error) Memory leak: digest
[src/gnutls_auth.cpp:159]: (error) Memory leak: digest
[src/gnutls_auth.cpp:161]: (error) Memory leak: digest
[src/gnutls_auth.cpp:174]: (error) Memory leak: digest
[src/gnutls_auth.cpp:179]: (error) Memory leak: digest
[src/gnutls_auth.cpp:181]: (error) Memory leak: digest
[src/gnutls_auth.cpp:111]: (error) Memory leak: digest
[src/gnutls_auth.cpp:120]: (error) Memory leak: digest
[src/gnutls_auth.cpp:130]: (error) Memory leak: digest
[src/gnutls_auth.cpp:139]: (error) Memory leak: digest
[src/nss_auth.cpp:128]: (error) Memory leak: digest
[src/nss_auth.cpp:132]: (error) Memory leak: digest
[src/string_util.cpp:307]: (error) Memory leak: result
[src/string_util.cpp:357]: (error) Memory leak: result
make: *** [cppcheck] Error 1
```

https://travis-ci.org/s3fs-fuse/s3fs-fuse/jobs/336597559#L1971

**Comment by ggtakec** [comment]
@gaul I'm sorry for my late reply.
I'm correcting error report by cppcheck on osx.
However, the problem is probably not solved with the latest version of cppcheck(brew, 1.82).
I'm still investigating, but phenomena similar to the following are out.
https://github.com/danmar/cppcheck/pull/1078
Please wait a moment.

**Comment by ggtakec** [comment]
Fixed this issue by #727.
I'm closing this, if you found a problem, please reopen this.
Thanks
